mediapipe/tasks/ios/core/sources/MPPTaskInfo.mm

```diff
@@ -0,0 +1,136 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "mediapipe/tasks/ios/core/sources/MPPTaskInfo.h"
+#import "mediapipe/tasks/ios/common/sources/MPPCommon.h"
+#import "mediapipe/tasks/ios/common/utils/sources/MPPCommonUtils.h"
+#import "mediapipe/tasks/ios/common/utils/sources/NSString+Helpers.h"
+
+#include "mediapipe/calculators/core/flow_limiter_calculator.pb.h"
+#include "mediapipe/framework/calculator.pb.h"
+#include "mediapipe/framework/calculator_options.pb.h"
+
+namespace {
+using CalculatorGraphConfig = ::mediapipe::CalculatorGraphConfig;
+using Node = ::mediapipe::CalculatorGraphConfig::Node;
+using ::mediapipe::InputStreamInfo;
+using ::mediapipe::CalculatorOptions;
+using ::mediapipe::FlowLimiterCalculatorOptions;
+}  // namespace
+
+@implementation MPPTaskInfo
+
+- (instancetype)initWithTaskGraphName:(NSString *)taskGraphName
+                         inputStreams:(NSArray<NSString *> *)inputStreams
+                        outputStreams:(NSArray<NSString *> *)outputStreams
+                          taskOptions:(id<MPPTaskOptionsProtocol>)taskOptions
+                   enableFlowLimiting:(BOOL)enableFlowLimiting
+                                error:(NSError **)error {
+  if (!taskGraphName || !inputStreams.count || !outputStreams.count) {
+    [MPPCommonUtils
+        createCustomError:error
+                 withCode:MPPTasksErrorCodeInvalidArgumentError
+              description:
+                  @"Task graph's name, input streams, and output streams should be non-empty."];
+  }
+
+  self = [super init];
+
+  if (self) {
+    _taskGraphName = taskGraphName;
+    _inputStreams = inputStreams;
+    _outputStreams = outputStreams;
+    _taskOptions = taskOptions;
+    _enableFlowLimiting = enableFlowLimiting;
+  }
+  return self;
+}
+
+- (id)copyWithZone:(NSZone *)zone {
+  MPPTaskInfo *taskInfo = [[MPPTaskInfo alloc] init];
+
+  taskInfo.taskGraphName = self.taskGraphName;
+  taskInfo.inputStreams = self.inputStreams;
+  taskInfo.outputStreams = self.outputStreams;
+  taskInfo.taskOptions = self.taskOptions;
+  taskInfo.enableFlowLimiting = self.enableFlowLimiting;
+
+  return taskInfo;
+}
+
+- (CalculatorGraphConfig)generateGraphConfig {
+  CalculatorGraphConfig graph_config;
+
+  Node *task_subgraph_node = graph_config.add_node();
+  task_subgraph_node->set_calculator(self.taskGraphName.cppString);
+  [self.taskOptions copyToProto:task_subgraph_node->mutable_options()];
+
+  for (NSString *outputStream in self.outputStreams) {
+    auto cpp_output_stream = std::string(outputStream.cppString);
+    task_subgraph_node->add_output_stream(cpp_output_stream);
+    graph_config.add_output_stream(cpp_output_stream);
+  }
+
+  if (self.enableFlowLimiting) {
```

Comment:
Can you invert this as well to follow the Android implementation more closely?

Reply:
Inverted